ocs_ci/deployment/baremetal.py

```diff
@@ -13,4 +27,394 @@ class BAREMETALUPI(Deployment):
+                  f"{constants.coreos_url_prefix}" \
+                  f"{image_data['installer_initramfs_url']}"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            # Download installer_kernel
+            cmd = f"wget -O {constants.bm_tftp_dir}/rhcos-installer-kernel-x86_64 " \
+                  f"{constants.coreos_url_prefix}" \
+                  f"{image_data['installer_kernel_url']}"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            # Download metal_bios
+            cmd = f"wget -O {constants.bm_path_to_upload}/rhcos-metal.x86_64.raw.gz " \
+                  f"{constants.coreos_url_prefix}" \
+                  f"{image_data['metal_bios_url']}"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to Download required File"
+            # Create pxelinux.cfg directory
+            cmd = f"mkdir -m 755 {constants.bm_tftp_dir}/pxelinux.cfg"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to create required folder"
+
+        def deploy(self, log_cli_level='DEBUG'):
+            """
+            Deploy
+            """
+            # Uploading pxe files
+            logger.info("Deploying OCP cluster for Bare Metal platform")
+            logger.info(
+                f"Openshift-installer will be using log level:{log_cli_level}"
+            )
+            upload_file(
+                self.host,
+                constants.COMMON_CONF_FILE,
+                os.path.join(
+                    constants.bm_dnsmasq_dir,
+                    "dnsmasq.common.conf"
+                ),
+                self.user,
+                key_file=self.private_key
+            )
+            file_list = os.listdir(constants.PXE_FILE)
+            for files in file_list:
+                upload_file(
+                    self.host,
+                    f"{constants.PXE_FILE}/{files}",
+                    os.path.join(
+                        f"{constants.bm_tftp_dir}/pxelinux.cfg",
+                        f"{files}"
+                    ),
+                    self.user,
+                    key_file=self.private_key
+                )
+            # Applying Permission
+            cmd = f"chmod 755 -R {constants.bm_tftp_dir}"
+            self.helper_node_handler.exec_cmd(cmd=cmd)
+
+            # Applying Permission
+            cmd = f"chmod 755 -R {constants.bm_path_to_upload}"
+            self.helper_node_handler.exec_cmd(cmd=cmd)
+
+            # Restarting dnsmasq service
+            cmd = "systemctl restart dnsmasq"
+            assert self.helper_node_handler.exec_cmd(cmd=cmd), "Failed to restart dnsmasq service"
+            # Rebooting Machine with pxe boot
+            self.mgmt_details = load_auth_config()['ipmi']
+            for machine_id in range(1, 8):
+                machine_name = f"argo00{machine_id}.ceph.redhat.com"
```

Comment:
any chance to save the machine name template in a conf file?

Reply:
machine name with roles will be added in auth.yaml

Reply:
Done

Reply:
in a new commit?